destination/franz_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright © 2023 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
+
+import (
+	"context"
+	"crypto/tls"
+	"testing"
+	"time"
+
+	"github.com/conduitio/conduit-connector-kafka/common"
+	"github.com/conduitio/conduit-connector-kafka/test"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/matryer/is"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/pkg/sasl"
+)
+
+func TestFranzProducer_Opts(t *testing.T) {
+	is := is.New(t)
+
+	clientCert, clientKey, caCert := test.Certificates(t)
+
+	cfg := Config{
+		Config: common.Config{
+			Servers:  []string{"test-host:9092"},
+			Topic:    "test-topic",
+			ClientID: "test-client-id",
+
+			ConfigSASL: common.ConfigSASL{
+				Mechanism: "PLAIN",
+				Username:  "user",
+				Password:  "pass",
+			},
+			ConfigTLS: common.ConfigTLS{
+				ClientCert: clientCert,
+				ClientKey:  clientKey,
+				CACert:     caCert,
+			},
+		},
+		BatchBytes:      512,
+		DeliveryTimeout: time.Second,
+		Acks:            "one",
+	}
+
+	p, err := NewFranzProducer(context.Background(), cfg)
+	is.NoErr(err)
+
+	is.Equal(p.client.OptValue(kgo.DefaultProduceTopic), cfg.Topic)
+	is.Equal(p.client.OptValue(kgo.AllowAutoTopicCreation), true)
+	is.Equal(p.client.OptValue(kgo.RecordDeliveryTimeout), cfg.DeliveryTimeout)
+	is.Equal(p.client.OptValue(kgo.RequiredAcks), cfg.RequiredAcks())
+	is.Equal(p.client.OptValue(kgo.DisableIdempotentWrite), true) // only true because Acks is "one"
```

Comment:
Nitpick: maybe have a test for other settings? It's probably low on the priority list.